src/rprblender/export/material.py

```diff
@@ -23,12 +23,9 @@
 
 def key(material: bpy.types.Material, obj=None, input_socket_key='Surface'):
     mat_key = material.name_full
-    if obj and obj.type == 'MESH' and has_uv_map_node(material):
-        mat_key = (mat_key, obj.data.rpr.uv_sets_names)
-    if input_socket_key != 'Surface':
-        mat_key = (mat_key, input_socket_key)
+    obj_name = obj.name_full if obj is not None else ''
 
-    return mat_key
+    return (mat_key, obj_name, input_socket_key)
```

Comment:
Can be simplified to: return (material.name_full, object.key(obj) if obj else None, input_socket_key)